File: pkg/dev_compiler/test/codegen_expected/language/allocate_large_object_test.js
dart_library.library('language/allocate_large_object_test', null, /* Imports */[
  'dart_sdk',
  'expect'
], function load__allocate_large_object_test(exports, dart_sdk, expect) {
  'use strict';
  const core = dart_sdk.core;
  const dart = dart_sdk.dart;
  const dartx = dart_sdk.dartx;
  const expect$ = expect.expect;
  const allocate_large_object_test = Object.create(null);
  let VoidTodynamic = () => (VoidTodynamic = dart.constFn(dart.definiteFunctionType(dart.dynamic, [])))();
  allocate_large_object_test.A = class A extends core.Object {
    static foo() {
      return allocate_large_object_test.A.s;
    }
    new(a) {
      this.a = a;
      this.d1 = null;
      this.d2 = null;
      this.d3 = null;
      this.d4 = null;
      this.d5 = null;
      this.d6 = null;
      this.d7 = null;
      this.d8 = null;
      this.d9 = null;
      this.d10 = null;
      this.d11 = null;
      this.d12 = null;
      this.d13 = null;
      this.d14 = null;
    }
    value() {
      return dart.notNull(this.a) + dart.notNull(core.num._check(allocate_large_object_test.A.foo()));
    }
  };
  dart.setSignature(allocate_large_object_test.A, {
    constructors: () => ({new: dart.definiteFunctionType(allocate_large_object_test.A, [core.int])}),
    methods: () => ({value: dart.definiteFunctionType(dart.dynamic, [])}),
    statics: () => ({foo: dart.definiteFunctionType(dart.dynamic, [])}),
    names: ['foo']
  });
  allocate_large_object_test.A.s = null;
  allocate_large_object_test.AllocateLargeObject = class AllocateLargeObject extends core.Object {
    static testMain() {
      let a = new allocate_large_object_test.A(1);
      allocate_large_object_test.A.s = 4;
      expect$.Expect.equals(5, a.value());
    }
  };
  dart.setSignature(allocate_large_object_test.AllocateLargeObject, {
    statics: () => ({testMain: dart.definiteFunctionType(dart.dynamic, [])}),
    names: ['testMain']
  });
  allocate_large_object_test.main = function() {
    allocate_large_object_test.AllocateLargeObject.testMain();
  };
  dart.fn(allocate_large_object_test.main, VoidTodynamic());
  // Exports:
  exports.allocate_large_object_test = allocate_large_object_test;
});

File: tests/language_2/allocate_large_object_test.dart
// Copyright (c) 2011, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

import "package:expect/expect.dart";

class A {
  int a;
  double d1;
  double d2;
  double d3;
  double d4;
  double d5;
  double d6;
  double d7;
  double d8;
  double d9;
  double d10;
  double d11;
  double d12;
  double d13;
  double d14;
  static var s;

  static foo() {
    return s;
  }

  A(this.a) {}

  value() {
    return a + foo();
  }
}

class AllocateLargeObject {
  static testMain() {
    var a = new A(1);
    A.s = 4;
    Expect.equals(5, a.value());
  }
}

main() {
  AllocateLargeObject.testMain();
}
